<?xml version="1.0"?>
<robot name="panda_mounted_husky">

  <!-- Virtual joint -->
  <virtual_joint name="virtual_joint"
                 type="fixed"
                 parent_frame="world"
                 child_link="base_link"/>

  <!-- Arm group: 7 DOF Panda arm -->
  <group name="panda_arm">
    <chain base_link="panda_link0" tip_link="panda_link7"/>
  </group>

  <!-- Gripper group -->
  <group name="panda_hand">
    <joint name="panda_finger_joint1"/>
    <joint name="panda_finger_joint2"/>
  </group>

  <!-- Combined arm+gripper -->
  <group name="panda_arm_hand">
    <group name="panda_arm"/>
    <group name="panda_hand"/>
  </group>

  <!-- End effector -->
  <end_effector name="panda_gripper"
                parent_link="panda_link7"
                group="panda_hand"/>

  <!-- Predefined states -->
  <group_state group="panda_arm" name="ready">
    <joint name="panda_joint1" value="0"/>
    <joint name="panda_joint2" value="-0.785"/>
    <joint name="panda_joint3" value="0"/>
    <joint name="panda_joint4" value="-2.356"/>
    <joint name="panda_joint5" value="0"/>
    <joint name="panda_joint6" value="1.571"/>
    <joint name="panda_joint7" value="0.785"/>
  </group_state>

  <group_state group="panda_hand" name="open">
    <joint name="panda_finger_joint1" value="0.04"/>
    <joint name="panda_finger_joint2" value="0.04"/>
  </group_state>

  <group_state group="panda_hand" name="close">
    <joint name="panda_finger_joint1" value="0.0"/>
    <joint name="panda_finger_joint2" value="0.0"/>
  </group_state>

  <!-- Disable adjacent collisions -->
  <disable_collisions link1="panda_link1" link2="panda_link2" reason="Adjacent"/>
  <disable_collisions link1="panda_link2" link2="panda_link3" reason="Adjacent"/>
  <disable_collisions link1="panda_link3" link2="panda_link4" reason="Adjacent"/>
  <disable_collisions link1="panda_link4" link2="panda_link5" reason="Adjacent"/>
  <disable_collisions link1="panda_link5" link2="panda_link6" reason="Adjacent"/>
  <disable_collisions link1="panda_link6" link2="panda_link7" reason="Adjacent"/>
  <disable_collisions link1="panda_link7" link2="panda_leftfinger" reason="Adjacent"/>
  <disable_collisions link1="panda_link7" link2="panda_rightfinger" reason="Adjacent"/>
  <disable_collisions link1="panda_leftfinger" link2="panda_rightfinger" reason="Adjacent"/>

</robot>
